### PR TITLE
kubectl: fixes issues #45736 and #45737

### LINKED ISSUE #45736: kubectl proxy --help is missing description of the command
kubectl Version: 1.6.2

**Problem:** `kubectl proxy --help` command directly jumps into the examples of the command without explaining what the command does. For example:

```
$ kubectl proxy -h
To proxy all of the kubernetes api and nothing else, use: 

  $ kubectl proxy --api-prefix=/
  
To proxy only part of the kubernetes api and also some static files: 

  $ kubectl proxy --www=/my/files --www-prefix=/static/ --api-prefix=/api/
  
The above lets you 'curl localhost:8001/api/v1/pods'. 

To proxy the entire kubernetes api at a different root, use: 

  $ kubectl proxy --api-prefix=/custom/
  
The above lets you 'curl localhost:8001/custom/api/v1/pods'

Examples:
  # Run a proxy to kubernetes apiserver on port 8011, serving static content from ./local/www/
  kubectl proxy --port=8011 --www=./local/www/
  
  # Run a proxy to kubernetes apiserver on an arbitrary local port.
  # The chosen port for the server will be output to stdout.
  kubectl proxy --port=0
  
  # Run a proxy to kubernetes apiserver, changing the api prefix to k8s-api
  # This makes e.g. the pods api available at localhost:8001/k8s-api/v1/pods/
  kubectl proxy --api-prefix=/k8s-api
...

```

Compare this with:

```
 kubectl apply -h
Apply a configuration to a resource by filename or stdin. This resource will be created if it doesn't exist yet. To use
'apply', always create the resource initially with either 'apply' or 'create --save-config'.

JSON and YAML formats are accepted.

Alpha Disclaimer: the --prune functionality is not yet complete. Do not use unless you are aware of what the current
state is. See https://issues.k8s.io/34274.

Examples:
[...]
```

```
$ kubectl rollout -h
Manage a deployment using subcommands like "kubectl rollout undo deployment/abc"

Examples:
[...]
```

### LINKED ISSUE #45737: kubectl proxy --help examples are unorganized and overlapping
kubectl version: 1.6.2.

**Problem:** help for `kubectl proxy`  has an `Examples:` section, but half of the examples section are in the main area where instead the command should be explained. #45736 Also the examples under the `Examples:` and before the `Examples:` are overlapping a bit.

```
To proxy all of the kubernetes api and nothing else, use: 

  $ kubectl proxy --api-prefix=/
  
To proxy only part of the kubernetes api and also some static files: 

  $ kubectl proxy --www=/my/files --www-prefix=/static/ --api-prefix=/api/
  
The above lets you 'curl localhost:8001/api/v1/pods'. 

To proxy the entire kubernetes api at a different root, use: 

  $ kubectl proxy --api-prefix=/custom/
  
The above lets you 'curl localhost:8001/custom/api/v1/pods'

Examples:
  # Run a proxy to kubernetes apiserver on port 8011, serving static content from ./local/www/
  kubectl proxy --port=8011 --www=./local/www/
  
  # Run a proxy to kubernetes apiserver on an arbitrary local port.
  # The chosen port for the server will be output to stdout.
  kubectl proxy --port=0
  
  # Run a proxy to kubernetes apiserver, changing the api prefix to k8s-api
  # This makes e.g. the pods api available at localhost:8001/k8s-api/v1/pods/
  kubectl proxy --api-prefix=/k8s-api

Options:
      --accept-hosts='^localhost$,^127\.0\.0\.1$,^\[::1\]$': Regular expression for hosts that the proxy should accept.
      --accept-paths='^.*': Regular expression for paths that the proxy should accept.
      --address='127.0.0.1': The IP address on which to serve on.
      --api-prefix='/': Prefix to serve the proxied API under.
      --disable-filter=false: If true, disable request filtering in the proxy. This is dangerous, and can leave you vulnerable to XSRF attacks, when used with an accessible port.
  -p, --port=8001: The port on which to run the proxy. Set to 0 to pick a random port.
      --reject-methods='POST,PUT,PATCH': Regular expression for HTTP methods that the proxy should reject.
      --reject-paths='^/api/.*/pods/.*/exec,^/api/.*/pods/.*/attach': Regular expression for paths that the proxy should reject.
  -u, --unix-socket='': Unix socket on which to run the proxy.
  -w, --www='': Also serve static files from the given directory under the specified prefix.
  -P, --www-prefix='/static/': Prefix to serve static files under, if static file directory is specified.

Usage:
  kubectl proxy [--port=PORT] [--www=static-dir] [--www-prefix=prefix] [--api-prefix=prefix] [options]

Use "kubectl options" for a list of global command-line options (applies to all commands).
```

### PR DESCRIPTION
**Pull request by robertojrojas**
**What this PR does / why we need it**:
This PR improves the help displayed by the `kubectl proxy` command.

**Which issue this PR fixes**: 
fixes #45736, fixes #45737 

**Special notes for your reviewer**:
NONE

**Release note**:
NONE

```release-note
```

### CODE REVIEW

**Comment by cblecker** [comment]
@k8s-bot ok to test
/release-note-none

File: pkg/kubectl/cmd/proxy.go [hunk 1]
```diff
@@ -67,7 +70,7 @@ var (
 func NewCmdProxy(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:     "proxy [--port=PORT] [--www=static-dir] [--www-prefix=prefix] [--api-prefix=prefix]",
-		Short:   i18n.T("Run a proxy to the Kubernetes API server"),
+		Short:   proxyLong,
```
Comment (chrislovecnm) [inline pkg/kubectl/cmd/proxy.go]: You need to leave short in and set Long to proxyLong.

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
Noticed you accidentally replaced short with long.

**Comment by cblecker** [comment]
@k8s-bot kops aws e2e test this
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by chrislovecnm** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Review by chrislovecnm — APPROVED** [review]
Looks great

**Comment by robertojrojas** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this

**Comment by ahmetb** [comment]
@robertojrojas it looks like #45902 was also opened after this, we can go with your PR. #45902 has a bit more detailed explanation, should we incorporate that in this PR?

**Comment by robertojrojas** [comment]
@ahmetb sure, no problem. I like that description better than what I have. I can take that description and update my PR. does that sound good to you?

**Comment by ahmetb** [comment]
@robertojrojas I am not sure about the correctness. Make sure somebody verifies its contents :)

**Comment by ahmetb** [comment]
Is there a way to stop this from getting merged?

**Comment by cblecker** [comment]
@ahmetb Yup :) `/lgtm cancel` does that. We're good.

**Comment by ahmetb** [comment]
@robertojrojas do you have time to follow up with the changes on this PR

**Comment by robertojrojas** [comment]
@ahmetb sorry, but I'm not sure I understand what you're asking me.

**Comment by ahmetb** [comment]
@robertojrojas so we cancelled merging above (normally this was approved) because we're waiting on you to add the detailed explanation for `kubectl proxy` in the pull request #45902.

**Comment by robertojrojas** [comment]
@ahmetb ah yeah. I'll do that today.

**Comment by robertojrojas** [comment]
@ahmetb @cblecker let me know if you need to change anything.

**Comment by ahmetb** [comment]
likely a flake
@k8s-bot kops aws e2e test this

**Comment by cblecker** [comment]
@robertojrojas The indentation is not quite right for lines 38 through 41. Can you ensure there are two tab characters proceeding each of those lines, with no white spaces?

**Comment by cblecker** [comment]
@robertojrojas Fantastic! One last thing before merge.. can you please swash the 4 commits into one?

**Comment by ahmetb** [comment]
@cblecker is this necessary? I'm trying to see if squashing commits is documented anywhere on the contributor guide.

**Comment by cblecker** [comment]
It isn't strictly necessary, but it provides a more useful history when the commits are all fixups based on feedback, rather than discrete features that we'd want a history for:
https://github.com/kubernetes/community/blob/master/contributors/devel/pull-requests.md#6-squashing-and-commit-titles

**Comment by cblecker** [comment]
@robertojrojas Thanks for your contribution! Once tests pass (don't worry about the federation test -- it's a known issue and is being worked on), the Submit Queue should automatically merge your PR.

/lgtm

**Comment by robertojrojas** [comment]
@cblecker happy to help!

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-e2e-kops-aws test this

**Comment by cblecker** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Pull request merged**